packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -336,20 +337,16 @@ abstract class FlutterCommand extends Command<void> {
         valueHelp: 'x.y.z');
   }
 
-  void usesDartDefines() {
+  void usesDartDefinesOption() {
     argParser.addMultiOption(
-      'dart-define',
+      FlutterOptions.kDartDefinesOption,
       help: 'Passed to the Dart compiler building this application as a -D flag.\n'
             'Values supported by this option are compiler implementation specific.\n'
```

Comment:
Is this true?

Should we note somewhere that the environment values are treated by the compiler as constants for the purposes of dead-code elimination, etc.?

Reply:
Updated the description